linera-execution/src/runtime.rs

```diff
@@ -1016,48 +1059,25 @@ impl ContractSyncRuntime {
             self.execute(application, context.authenticated_signer, |contract| {
                 contract.finalize(context)
             })?;
+            self.inner().loaded_applications.remove(&application);
```

Comment:
Aren't we dropping a loaded application that owns a `SyncRuntime` in the first iteration? Won't dropping that clear `loaded_applications`?

Reply:
Not really, because the application instance owns a `SyncRuntimeHandle` instead.